File: Behavioral/State/state.cs
interface State
{
    void Handle(Context context);
}

class Context
{
    public State State;
    public Context(State state) => State = state;
    public void Request() => State.Handle(this);
}

class OnState : State
{
    public void Handle(Context context)
    {
        context.State = new OffState();
    }
}

class OffState : State
{
    public void Handle(Context context)
    {
        context.State = new OnState();
    }
}

File: Behavioral/State/state.py
class State:
    def handle(self, context):
        pass

class Context:
    def __init__(self, state):
        self.state = state

    def request(self):
        self.state.handle(self)

class OnState(State):
    def handle(self, context):
        context.state = OffState()

class OffState(State):
    def handle(self, context):
        context.state = OnState()
